File: My algorithms collection/Tower of Hanoi/tower_of_hanoi.c
#include <stdio.h>
void towerofhanoi(int n, char source, char auxilliary, char target){
    if (n==1){
        printf("move %dth from %c to %c\n", n, source, target);
        return;
        
    }
    towerofhanoi(n-1, source, target, auxilliary); //move n-1 to auxiliary
    printf("move %dth from %c to %c\n", n, source, target); //move nth to target
    towerofhanoi(n-1, auxilliary, source, target); //move n-1 to target
}
    
    
    int main()
{
    char source, auxilliary, target;
    int n;
    printf("what will you like to denote source with: \n");
    scanf(" %c", &source);
    printf("what will you like to denote auxilliary with: \n");
    scanf(" %c", &auxilliary);
    printf("what will you like to denote target with: \n");
    scanf(" %c", &target);
    printf("how many disks are in the source?: ");
    scanf(" %d", &n);
    towerofhanoi(n, source, auxilliary, target);

    return 0;
}

File: My algorithms collection/Tower of Hanoi/tower_of_hanoi.js
function towerOfHanoi(n, source, auxiliary, target) {
    if (n === 1) {
        console.log(`Move disk 1 from ${source} to ${target}`);
        return;
    }
    towerOfHanoi(n - 1, source, target, auxiliary); //moves the n-1 disks to auxiliary
    console.log(`Move disk ${n} from ${source} to ${target}`); //moves n to target
    towerOfHanoi(n - 1, auxiliary, source, target); //moves n-1 from auxiliary to target
}

let source = prompt("What will you like to denote source with?");
let auxiliary = prompt("What will you like to denote auxiliary with?");
let target = prompt("What will you like to denote target with?");
let n = parseInt(prompt("How many disks are in the source?"));
towerOfHanoi(n, source, auxiliary, target);
